### PR TITLE
3685: Remove zio's Duration and replace it with java.time.Duration

### PR DESCRIPTION
**Pull request by adrianfilip**

### CODE REVIEW

**Comment by adrianfilip** [comment]
This PR is for https://github.com/zio/zio/issues/3685

File: core-tests/jvm/src/test/scala-2.12/zio/StacktracesSpec.scala [hunk 1]
```diff
@@ -54,7 +54,7 @@ object StackTracesSpec extends DefaultRunnableSpec {
     },
     testM("foreachPar fail") {
       val io = for {
-        trace <- TestClock.setTime(1.second) *> foreachParFail
+        trace <- TestClock.setTime(fromSeconds(1)) *> foreachParFail
```
Comment (jdegoes) [inline core-tests/jvm/src/test/scala-2.12/zio/StacktracesSpec.scala:57]: Let's have extension methods in `zio.duration._` so we can still use the same syntax as before. Will help ensure people don't have to change their code.

File: core-tests/shared/src/test/scala/zio/ZIOSpec.scala [hunk 2]
```diff
@@ -1954,16 +1954,16 @@ object ZIOSpec extends ZIOBaseSpec {
...
       },
       testM("timeout repetition of uninterruptible effect") {
         val effect = ZIO.unit.uninterruptible.forever
 
-        assertM(Live.live(effect.timeout(1.second)))(isNone)
+        assertM(Live.live(effect.timeout(fromSeconds(1))))(isNone)
       } @@ jvmOnly @@ zioTag(interruption),
       testM("timeout in uninterruptible region") {
-        val effect = (ZIO.unit.timeout(Duration.Infinity)).uninterruptible
+        val effect = (ZIO.unit.timeout(zio.duration.infiniteDuration)).uninterruptible
```
Comment (jdegoes) [inline core-tests/shared/src/test/scala/zio/ZIOSpec.scala:1966]: We can still keep `Duration.Infinity` / etc. There can be a type alias `Duration` inside `zio.duration` which delegates to `java.time.Duration`, then there can be a companion object with vals inside it and helpers. So overall almost no code should change in user-land.

File: core-tests/jvm/src/test/scala-2.12/zio/StacktracesSpec.scala (hunk 1 above)
Comment (adrianfilip) [reply core-tests/jvm/src/test/scala-2.12/zio/StacktracesSpec.scala:57]: Good point, we can still use extension methods.

**Comment by adrianfilip** [comment]
Closing this because I opened a new one: https://github.com/zio/zio/pull/4029

**Comment by adrianfilip** [comment]
https://github.com/zio/zio/pull/4029

**Pull request closed without merging**